libraries/SRV_Channel/SRV_Channel_config.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifndef NUM_SERVO_CHANNELS
+#if defined(HAL_BUILD_AP_PERIPH) && defined(HAL_PWM_COUNT)
```

Comment:
You're in-between two styles making it a little unintuitive, please pick one:

```
#ifndef NUM_SERVO_CHANNELS
#if defined(HAL_BUILD_AP_PERIPH) && defined(HAL_PWM_COUNT)
    #define NUM_SERVO_CHANNELS HAL_PWM_COUNT
#elif defined(HAL_BUILD_AP_PERIPH)
    #define NUM_SERVO_CHANNELS 0
#elif !HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024
    #define NUM_SERVO_CHANNELS 32
#else
    #define NUM_SERVO_CHANNELS 16
#endif
#endif
```
 --- or ---
 ```
#ifndef NUM_SERVO_CHANNELS
#if defined(HAL_BUILD_AP_PERIPH)
    #if defined(HAL_PWM_COUNT)
        #define NUM_SERVO_CHANNELS HAL_PWM_COUNT
    #else
        #define NUM_SERVO_CHANNELS 0
    #endif
#else
    #if !HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024
        #define NUM_SERVO_CHANNELS 32
    #else
        #define NUM_SERVO_CHANNELS 16
    #endif
#endif
 ```

Reply:
@magicrub I was planning on coming back through here and moving those AP_PERIPH bits down into `chibios_hwdef.py`.

Given @joshanne is just moving this code from one place to another, are you OK with leaving the style as-is for now?  (i.e. the formatting is not his fault! :-)

Reply:
@peterbarker yeah, that's fine. I didn't realize it was actually a pure copy/paste on this one. For some reason I thought it was coming from the other block that said:
```
#ifdef NUM_SERVO_CHANNELS
 #define ESC_TELEM_MAX_ESCS NUM_SERVO_CHANNELS
#elif !HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024
 #define ESC_TELEM_MAX_ESCS 32
#else
 #define ESC_TELEM_MAX_ESCS 16
#endif
#define ESC_TELEM_MAX_ESCS NUM_SERVO_CHANNELS```